applications/StructuralMechanicsApplication/custom_elements/cr_beam_element_3D2N.hpp

```diff
@@ -286,33 +270,41 @@ class CrBeamElement3D2N : public Element
 
     /**
      * @brief This function calculates the symmetric deformation modes
-     * @param VectorDifference The vector differences of the quaternions
      */
-    Vector CalculateSymmetricDeformationMode(const Vector& VectorDifference) const;
+    Vector CalculateSymmetricDeformationMode();
 
     /**
      * @brief This function calculates the antisymmetric deformation modes
-     * @param Bisectrix The bisectrix between the local axis1 from the last iter. step and the updated axis 1
      */
-    Vector CalculateAntiSymmetricDeformationMode(const Vector& Bisectrix) const;
+    Vector CalculateAntiSymmetricDeformationMode();
 
     /**
      * @brief This function calculates the local nodal forces
      * @param Bisectrix The bisectrix between the local axis1 from the last iter. step and the updated axis 1
      * @param VectorDifference The vector differences of the quaternions
      */
-    void CalculateLocalNodalForces(const Vector& Bisectrix,const Vector& VectorDifference);
+    Vector CalculateLocalNodalForces();
+
+    void FinalizeNonLinearIteration(ProcessInfo& rCurrentProcessInfo) override;
+
+    Vector CalculateGlobalNodalForces();
+
+    Vector GetIncrementDeformation() const;
+
+    BoundedMatrix<double, msElementSize, msElementSize> GetTransformationMatrixGlobal();
+
+    void InitializeSolutionStep(ProcessInfo& rCurrentProcessInfo) override;
+    void InitializeNonLinearIteration(ProcessInfo& rCurrentProcessInfo) override;
 
 private:
 
-    int mIterationCount = 0;
-    Vector mTotalNodalDeformation = ZeroVector(msElementSize); // save as the displacement from the last iteration step is needed
-    Matrix mLocalRotationMatrix  = ZeroMatrix(msDimension); // save this as updating the matrix takes rather long
+
+    Vector mDeformationCurrent = ZeroVector(msElementSize);

```

Comment:
I understand you save the previous, but the current can be computed at the moment (even the previous using the previous disp and rot)